certbot-apache/certbot_apache/configurator.py

```diff
@@ -875,6 +885,10 @@ def make_vhost_ssl(self, nonssl_vhost):  # pylint: disable=too-many-locals
         # Create the Vhost object
         ssl_vhost = self._create_vhost(vh_p)
         ssl_vhost.ancestor = nonssl_vhost
+
+        if self.conf("handle-sites") and "/sites-enabled/" not in ssl_vhost.filep:
+            # Set vhost as disabled for enabling later on
+            ssl_vhost.enabled = False
```

Comment:
This will be addressed in the distro specific reliability work

Reply:
I think we may want to address it now. On Debian based systems, we're incorrectly marking vhosts as enabled which can cause problems later in execution.

For example, while it's not trivial to repro, starting with the default Apache config on Ubuntu 16.04, do the following:
```
sudo a2dissite 000-default.conf
sudo mkdir /etc/apache2/sites-available2
sudo cp /etc/apache2/sites-available/000-default.conf /etc/apache2/sites-available2/
# Add a ServerName directive to /etc/apache2/sites-available2/000-default.conf that you can obtain a cert for
sudo certbot -d foo.science --apache --no-redirect --apache-vhost-root /etc/apache2/sites-available2/
# Add a ServerAlias directive to /etc/apache2/sites-available2/000-default.conf
sudo certbot -d foo.science --apache --redirect --apache-vhost-root /etc/apache2/sites-available2/
```
On this branch, this last command fails because it things the disabled vhost in `/etc/apache2/sites-available2` would conflict with a redirect vhost. If you checkout `master`, this problem goes away.